### ISSUE TITLE
1.3.4 resullts

### ISSUE THREAD

**Issue by gaykan** [idanwe/ionic-cli-meteor-whatsapp-tutorial#5]
After 1.3.4 get the following messages with no display. Using 1.3.2 or 1.3.4 does now make any change.

> [Error] Error: argument 2 must be an object
>     autorun
>     (anonymous function)
>     invoke
>     (anonymous function)
>     forEach
>     createInjector
>     doBootstrap
>     bootstrap
>     angularInit
>     (anonymous function)
>     trigger
>     eventHandler
> [Error] Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
> Watchers fired in the last 5 iterations: []
> http://errors.angularjs.org/1.4.3/$rootScope/infdig?p0=10&p1=%5B%5D
>     $digest
>     (anonymous function)
>     completeOutstandingRequest
>     (anonymous function)
> [Error] Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
> Watchers fired in the last 5 iterations: []
> http://errors.angularjs.org/1.4.3/$rootScope/infdig?p0=10&p1=%5B%5D
> http://localhost:8100/lib/ionic/js/ionic.bundle.js:13248:32
> $digest@http://localhost:8100/lib/ionic/js/ionic.bundle.js:28901:35
> $apply@http://localhost:8100/lib/ionic/js/ionic.bundle.js:29131:31
> done@http://localhost:8100/lib/ionic/js/ionic.bundle.js:23544:53
> completeRequest@http://localhost:8100/lib/ionic/js/ionic.bundle.js:23716:15
> requestLoaded@http://localhost:8100/lib/ionic/js/ionic.bundle.js:23657:24
>     (anonymous function)
>     (anonymous function)
>     $apply
>     done
>     completeRequest
>     requestLoaded

**Comment by NBAlex** [comment]
I have exactly the same error (Chapter 3 - Ionic WhatsApp Tutorial) using both versions 1.3.2 and 1.3.4 of the angular-meteor library. I have started project from scratch as well as did checkout at chapter 4 using the code provided on the github repo, but the problem is the same.

This error was thrown by _angular-meteor.bundle.js_, at the line **10666**:

``` javascript
if (!_.isObject(options)) throw Error('argument 2 must be an object');  
```

and the reason for throwing this error is the code block at the lines **9255-9313**, in the file _angular-meteor.bundle.js_:

``` javascript
var Processor = function Processor(options){                                                                       // 3430
    this.selfOptions = options;                                                                                       // 3431
    this.pipes = {};                                                                                                  // 3432
};                                                                                                                 // 3433
                                                                                                                   // 3434
Processor.prototype.options = function(options) {                                                                  // 3435
    if (options) {                                                                                                    // 3436
        this.selfOptions = options;                                                                                      // 3437
    }                                                                                                                 // 3438
    return this.selfOptions;                                                                                          // 3439
};                                                                                                                 // 3440
                                                                                                                   // 3441
Processor.prototype.pipe = function(name, pipe) {                                                                  // 3442
    if (typeof name === 'string') {                                                                                   // 3443
        if (typeof pipe === 'undefined') {                                                                               // 3444
            return this.pipes[name];                                                                                        // 3445
        } else {                                                                                                         // 3446
            this.pipes[name] = pipe;                                                                                        // 3447
        }                                                                                                                // 3448
    }                                                                                                                 // 3449
    if (name && name.name) {                                                                                          // 3450
        pipe = name;                                                                                                     // 3451
        if (pipe.processor === this) { return pipe; }                                                                    // 3452
        this.pipes[pipe.name] = pipe;                                                                                    // 3453
    }                                                                                                                 // 3454
    pipe.processor = this;                                                                                            // 3455
    return pipe;                                                                                                      // 3456
};                                                                                                                 // 3457
                                                                                                                   // 3458
Processor.prototype.process = function(input, pipe) {                                                              // 3459
    var context = input;                                                                                              // 3460
    context.options = this.options();                                                                                 // 3461
    var nextPipe = pipe || input.pipe || 'default';                                                                   // 3462
    var lastPipe, lastContext;                                                                                        // 3463
    while (nextPipe) {                                                                                                // 3464
        if (typeof context.nextAfterChildren !== 'undefined') {                                                          // 3465
            // children processed and coming back to parent                                                                 // 3466
            context.next = context.nextAfterChildren;                                                                       // 3467
            context.nextAfterChildren = null;                                                                               // 3468
        }                                                                                                                // 3469
                                                                                                                   // 3470
        if (typeof nextPipe === 'string') {                                                                              // 3471
            nextPipe = this.pipe(nextPipe);                                                                                 // 3472
        }                                                                                                                // 3473
        nextPipe.process(context);                                                                                       // 3474
        lastContext = context;                                                                                           // 3475
        lastPipe = nextPipe;                                                                                             // 3476
        nextPipe = null;                                                                                                 // 3477
        if (context) {                                                                                                   // 3478
            if (context.next) {                                                                                             // 3479
                context = context.next;                                                                                        // 3480
                nextPipe = lastContext.nextPipe || context.pipe || lastPipe;                                                   // 3481
            }                                                                                                               // 3482
        }                                                                                                                // 3483
    }                                                                                                                 // 3484
    return context.hasResult ? context.result : undefined;                                                            // 3485
};                                                                                                                 // 3486
                                                                                                                   // 3487
exports.Processor = Processor;  
```

Unless this problem is resolved, it's not possible at all to continue with the existing tutorial as well as to force Ionic to work nicely with the Meteor.

If I can provide you with further information please let me know.

**Comment by idanwe** [comment]
@gaykan @NBAlex  I'm very busy this days, hope to get some time through the weekend to help you

**Comment by NBAlex** [comment]
@idanwe that's so kind of you...thank you very much :+1:

**Comment by johnmarty** [comment]
As a work around to get it working you can try these steps:
- delete the www/lib dir
- edit the bower.js file removing all the ~ from the dep versions
- reinstall bower `$ bower i`

This did the trick for me. I was left with these bower versions:

```
$ bower list
bower check-new     Checking for new versions of the project dependencies...
HelloIonic ~/Development/angular-meteor/ionic-cli-meteor-whatsapp-tutorial
├─┬ accounts-phone#0.0.21
│ ├─┬ accounts-base-client-side#0.1.1
│ │ └── meteor-client-side#0.2.3 (latest is 1.2.0)
│ └── meteor-client-side#0.2.3 (latest is 1.2.0)
├─┬ angular-meteor#1.3.2 (latest is 1.3.4)
│ ├── angular#1.4.8 incompatible with 1.4.4 (1.4.4 available, latest is 1.5.0-rc.0)
│ └── meteor-client-side#0.2.3 (latest is 1.2.0)
├─┬ angular-moment#0.10.3 (latest is 1.0.0-beta.3)
│ ├── angular#1.4.8 (1.4.9-build.3+sha.8120ab2 available, latest is 1.5.0-rc.0)
│ └── moment#2.10.6 (latest is 2.11.0)
├─┬ ionic#1.1.0 (latest is 1.2.4)
│ ├── angular#1.4.8 incompatible with 1.4.3 (1.4.3 available, latest is 1.5.0-rc.0)
│ ├─┬ angular-animate#1.4.3 (latest is 1.5.0-rc.0)
│ │ └── angular#1.4.8
│ ├─┬ angular-sanitize#1.4.3 (latest is 1.5.0-rc.0)
│ │ └── angular#1.4.8
│ └─┬ angular-ui-router#0.2.13 (latest is 0.2.15)
│   └── angular#1.4.8 (1.5.0-rc.0 available)
├── meteor-client-side#0.2.3 (latest is 1.2.0)
└── moment#2.10.6 (latest is 2.11.0)
```

![screen shot 2016-01-08 at 6 52 54 am](https://cloud.githubusercontent.com/assets/4492480/12197373/8aa9f9f6-b5d4-11e5-8654-2b38c32a53a4.png)

**Comment by NBAlex** [comment]
@johnmarty I've tried as you suggested, but the same error still persist...nothing helps. These are my specs:

``` javascript
bower -v
1.7.2

node -v
v5.3.0

npm -v
3.3.12

ionic -v
1.7.12

meteor --version
Meteor 1.2.1
```

``` javascript
bower list
bower check-new     Checking for new versions of the project dependencies...
HelloIonic /..path../whatsapp
├─┬ angular-meteor#1.3.4
│ ├── angular#1.4.4 (latest is 1.5.0-rc.0)
│ └── meteor-client-side#0.2.3 (latest is 1.2.0)
├─┬ ionic#1.1.1 (latest is 1.2.4)
│ ├── angular#1.4.4 incompatible with 1.4.3 (1.4.3 available, latest is 1.5.0-rc.0)
│ ├─┬ angular-animate#1.4.3 (latest is 1.5.0-rc.0)
│ │ └── angular#1.4.4
│ ├─┬ angular-sanitize#1.4.3 (latest is 1.5.0-rc.0)
│ │ └── angular#1.4.4
│ └─┬ angular-ui-router#0.2.13 (latest is 0.2.15)
│   └── angular#1.4.4 (1.5.0-rc.0 available)
├── meteor-client-side#0.2.3 (latest is 1.2.0)
└── moment#2.11.0
```

bower.json

``` javascript
{
  "name": "HelloIonic",
  "private": "true",
  "devDependencies": {
    "ionic": "driftyco/ionic-bower#1.1.1"
  },
  "dependencies": {
    "moment": "2.11.0",
    "meteor-client-side": "0.2.3",
    "angular-meteor": "1.3.4"
  }
}
```

package.json

``` javascript
{
  "name": "ionic-project",
  "version": "1.1.1",
  "description": "An Ionic project",
  "dependencies": {
    "gulp": "^3.5.6",
    "gulp-sass": "^2.0.4",
    "gulp-concat": "^2.2.0",
    "gulp-minify-css": "^0.3.0",
    "gulp-rename": "^1.2.0"
  },
  "devDependencies": {
    "bower": "^1.7.2",
    "gulp-util": "^2.2.14",
    "shelljs": "^0.3.0"
  },
  "cordovaPlugins": [
    "cordova-plugin-device",
    "cordova-plugin-console",
    "cordova-plugin-whitelist",
    "cordova-plugin-splashscreen",
    "cordova-plugin-statusbar",
    "ionic-plugin-keyboard"
  ],
  "cordovaPlatforms": []
}
```

P.S.
I've used **_ionic  serve**_ to run the app

**Comment by DAB0mB** [comment]
I fixed it in the following pr https://github.com/Urigo/angular-meteor/pull/1078

**Comment by elmor34** [comment]
@johnmarty thanks. removing ~solved the problem with 1.3.2

@DAB0mB But my problem is with 1.3.4.  And it is not working.

Because I want to have a regular email authentication also.
And **accounts-password** is not working with 1.3.2. So I need to use 1.3.4.

I get the following eroor with 1.3.4, without adding no additional packages :

> **Error: argument 2 must be an object**
> autorunprocessor.js:61
> (anonymous function)processor.js:61
> invokeionic.bundle.js:17629
> (anonymous function)ionic.bundle.js:17451:85
> forEachionic.bundle.js:13515
> createInjectorionic.bundle.js:17451
> doBootstrapionic.bundle.js:14809
> bootstrapionic.bundle.js:14830
> angularInitionic.bundle.js:14724
> (anonymous function)ionic.bundle.js:41538
> triggerionic.bundle.js:16175
> eventHandlerionic.bundle.js:16450

**Comment by Urigo** [comment]
THIS REPO IS DEPRECATED

The repo has been moved to https://github.com/Urigo/IonicCLI-Meteor-WhatsApp.
